src/OpenKNX/Time/TimeManager.cpp

```diff
@@ -0,0 +1,732 @@
+                tm.tm_mon = stoi(cmd.substr(5, 2)) - 1;
+                tm.tm_mday = stoi(cmd.substr(7, 2));
+                tm.tm_hour = stoi(cmd.substr(9, 2));
+                tm.tm_min = stoi(cmd.substr(11, 2));
+            }
+            else
+            {
+                logInfoP("Invalid time format");
+                return;
+            }
+            calculateAndSetDstFlag(tm);
+            setLocalTime(tm, millis());
+        }
+#endif
+        bool TimeManager::processCommand(std::string& cmd, bool diagnoseKo)
+        {
+            if (cmd.rfind("tm") == 0)
+            {
+                if (cmd == "tm")
+                {
+                    commandInformation();
+                    return true;
+                }
+#ifdef OPENKNX_TIME_TESTCOMMAND
+                if (cmd == "tm test")
+                {
+                    commandTest();
+                    return true;
+                }
+#endif
+#ifdef OPENKNX_TIME_DIGAGNOSTIC
+                if (cmd == "tm ?")
+                {
+                    commandHelp();
+                    return true;
+                }
+                if (cmd == "tm setdst")
+                {
+                    setDaylightSavingMode(DaylightSavingMode::AlwaysDayLightSavingTime);
+                    return true;
+                }
+                if (cmd == "tm setst")
+                {
+                    setDaylightSavingMode(DaylightSavingMode::AlwaysStandardTime);
+                    return true;
+                }
+                if (cmd == "tm calcdst")
+                {
+                    setDaylightSavingMode(DaylightSavingMode::Calculated);
+                    return true;
+                }
+
+                if (cmd.rfind("tm ") == 0) // must be the last command check
+                {
+                    commandSetDateTime(cmd);
+                    return true;
+                }
+#endif
+            }
+            return false;
+        }
+
+        int TimeManager::daylightSavingTimeOffset()
+        {
+            return _dayLightSavingTimeOffset;
+        }
+
+        void TimeManager::setTimeProvider(TimeProvider* timeProvider)
+        {
+            if (_timeProvider != nullptr)
+                delete _timeProvider;
+            _timeProvideSupportKnxDaylightSavingTimeSwitch = false;
+            _timeProvider = timeProvider;
+            if (timeProvider != nullptr)
+                _timeProvideSupportKnxDaylightSavingTimeSwitch = timeProvider->supportKnxDaylightSavingTimeSwitch();
+            if (_configured && _timeProvider != nullptr)
+                _timeProvider->setup();
+            _waitTimerReadKo = 0;
+#ifdef BASE_KoIsSummertime
+            if (!ParamBASE_InternalTime && _timeProvideSupportKnxDaylightSavingTimeSwitch && ParamBASE_SummertimeAll == 0 /*Kommunikationsobjekt 'Sommerzeit aktiv'*/ && ParamBASE_ReadTimeDate)
+                _waitTimerReadKo = max(millis(), 1UL) - TimeProviderKnx::InitialReadDelayInMs;
```

Comment:
Sollte das vielleicht so aussehen?
```suggestion
                _waitTimerReadKo = max(millis() - TimeProviderKnx::InitialReadDelayInMs, 1UL);
```
Wobei das für milis()<TimeProviderKnx::InitialReadDelayInMs (nach Start oder Überlauf) auch noch nicht sauber funktionieren würde.

Reply:
Nein, ich stelle nur sicher, das _waitTimerReadKo nicht 0 ist, weil 0 eine Marker für nicht verwendet ist.

Reply:
Habe das nun refactored, das ist so ohnedies sehr schlecht lesbar. Jetzt prüfe ich beim DelayCheck gegen die richtige Zeitspanne.